ir/expression.def

```diff
@@ -66,10 +66,12 @@ class Sub : Operation_Binary {
 class Shl : Operation_Binary {
     stringOp = "<<";
     precedence = DBPrint::Prec_Shl;
+    Shl { if (type->is<Type::Unknown>() && left) type = left->type; }
```

Comment:
I don't think that this logic belongs to the IR class; it is maybe part of a typeChecker.